src/sampler/sampler.jl

```diff
@@ -58,45 +58,36 @@ function Random.rand!(rng::AbstractRNG, s::SimpleOPFSampler, data::Dict)
 end
 
 function _set_loads!(data, pd, qd)
-    L = length(data["load"])
+    L = data.L
     length(pd) == L || throw(DimensionMismatch())
     length(qd) == L || throw(DimensionMismatch())
     
-    for i in 1:L
-        ldat = data["load"]["$i"]
-        ldat["pd"] = pd[i]
-        ldat["qd"] = qd[i]
-    end
+    data.pd = pd
+    data.qd = qd
```

Comment:
You can use in-place modification here
```suggestion
    data.pd .= pd
    data.qd .= qd
```

(which BTW is a non-mutable operation for `data`)